src/common/keccak/keccak1600/amd64/bmi1/keccakf1600.jinc

```diff
@@ -26,13 +26,18 @@ inline fn __theta_rol_bmi1(reg u64[5] c) -> reg u64[5]
 {
   inline int x;
   reg u64[5] d;
+  reg u64 t;
 
   for x = 0 to 5
   { // D[x] = C[x + 1]
     d[x] = c[(x + 1) % 5];
 
     // D[x] = ROT(D[x], 1)
-    _, _, d[x] = #ROL_64(d[x], 1);
+    // ROL is not DOIT, so use shifts.
+    t = d[x];
+    ?{}, t = #SHL_64(t, 1);
+    ?{}, d[x] = #SHR_64(d[x], 63);
+    ?{}, d[x] = #OR(t, d[x]);
```

Comment:
Why don’t you use high-level operators like `<<`, `>>`, and `|`? It would make the code much more readable, I think.